### PR TITLE
Update debug

### PR DESCRIPTION
**Pull request by juanvallejo**
/assign deads2k

Pick of https://github.com/openshift/origin/pull/20215 on latest master.

### CODE REVIEW

File: pkg/oc/cli/cmd/debug.go [hunk 1]
```diff
@@ -269,7 +269,8 @@ func (o *DebugOptions) Complete(cmd *cobra.Command, f kcmdutil.Factory, args []s
 
 	// TODO: below should be turned into a method on PrintFlags
 	if (o.PrintFlags.OutputFormat != nil && len(*o.PrintFlags.OutputFormat) > 0) ||
-		(o.PrintFlags.TemplatePrinterFlags != nil && o.PrintFlags.TemplatePrinterFlags.TemplateArgument != nil) {
+		(o.PrintFlags.TemplatePrinterFlags != nil && o.PrintFlags.TemplatePrinterFlags.TemplateArgument != nil &&
+			len(*o.PrintFlags.TemplatePrinterFlags.TemplateArgument) > 0) {
```
Comment (juanvallejo) [inline pkg/oc/cli/cmd/debug.go]: @deads2k Updated this to also take into account when a template arg has been specified. Was making debug.sh fail

Comment (deads2k) [reply pkg/oc/cli/cmd/debug.go]: > @deads2k Updated this to also take into account when a template arg has been specified. Was making debug.sh fail

break these apart into bool assignments so I can read them here without the nesting.

Comment (juanvallejo) [reply pkg/oc/cli/cmd/debug.go]: Done

File: pkg/oc/cli/cmd/debug.go [hunk 2]
```diff
@@ -267,9 +267,14 @@ func (o *DebugOptions) Complete(cmd *cobra.Command, f kcmdutil.Factory, args []s
...
+	templateArgSpecified := o.PrintFlags.TemplatePrinterFlags != nil &&
+		o.PrintFlags.TemplatePrinterFlags.TemplateArgument != nil &&
+		len(*o.PrintFlags.TemplatePrinterFlags.TemplateArgument) > 0
+
+	outputFormatSpecified := o.PrintFlags.OutputFormat != nil && len(*o.PrintFlags.OutputFormat) > 0
+
 	// TODO: below should be turned into a method on PrintFlags
-	if (o.PrintFlags.OutputFormat != nil && len(*o.PrintFlags.OutputFormat) > 0) ||
-		(o.PrintFlags.TemplatePrinterFlags != nil && o.PrintFlags.TemplatePrinterFlags.TemplateArgument != nil) {
+	if outputFormatSpecified || templateArgSpecified {
```
Comment (deads2k) [inline pkg/oc/cli/cmd/debug.go]: what a bizarre command.

**Comment by juanvallejo** [comment]
@deads2k had to rebase to get completions updated / test verify passing

**Pull request merged**